Add more information on integration tests (#92)
konflux-ci/docs: docs/modules/ROOT/pages/glossary/index.adoc

:source-highlighter: highlightjs

**{ProductName}** +
A platform to automate the process of building, testing, and releasing applications. {ProductName} offers enterprise-grade security and customizable feature sets.

**build pipeline** +
A Tekton PipelineRun which converts from source code to a built and tested artifact in a container registry.

**build pipeline customization** +
The ability to update and manage build pipelines for each component in an application.

**build-time tests** +

> A specific TaskRun which is included in a build pipeline but does not required to convert from the source control to the produced artifact. This may include any number of functional or security checks produced on the source code or the produced artifact.

Reviewer: Is it source control or source code?

How about this? Does it change the meaning?
```suggestion
A specific TaskRun within in a build pipeline that doesn't convert source code into  product artifact can include any number of functional or security checks performed on the source code or the produced artifact.
```

Author: It is source code in source control. Using source code might be clearer though.

**cluster** +
A Kubernetes deployment with nodes that run containerized applications and a control plane that manages the nodes.

**component** +

> An internal representation of a software artifact that {ProductName} builds and tests, using source code from a git repository. Components are stored an link:https://github.com/opencontainers/distribution-spec[OCI container registry] such as quay.io after they are built.

Reviewer: ```suggestion
An internal representation of a software artifact that {ProductName} builds and tests, using source code from a git repository. Components are stored on link:https://github.com/opencontainers/distribution-spec[OCI container registry], such as quay.io, after they are built.
```

**Enterprise Contract (EC)** +
A set of release policies that you apply to your application snapshot. You can use the EC to prevent releases that are not compliant with Supply-chain Levels for Software Artifacts (SLSA) guidelines. 

**integration test** +

> A pipeline defined in source control that is run after the completion of a component's build pipeline. Integration tests are run against all components in a snapshot as a whole.

Reviewer 2: ```suggestion
A Tekton pipeline defined in source control that is run after the completion of a component's build pipeline. Integration tests are run against all components in a snapshot as a whole.
```

> A pipeline defined in source control that is run after the completion of a component's build pipeline. Integration tests are run against all components in a snapshot as a whole.

Reviewer: ```suggestion
A pipeline defined in source control, runs after the components are built. It performs the integration against all components in a snapshot as a whole.
```

**IntegrationTestScenario (ITS)** +

> A Kubernetes resource that defines a pipeline to run for an integration test. The integration service uses the ITS to test application components.

Reviewer 2: Technically the ITS just defines metadata about the test and contains a resolver for a tekton pipeline.  We should make that clear here so that users understand how their tests are run.

Author: Are you saying that you suggestion is making it clearer? The information that you commented feels like too much for the documentation.

> A Kubernetes resource that defines a pipeline to run for an integration test. The integration service uses the ITS to test application components.

Reviewer: ```suggestion
A Kubernetes resource that defines a pipeline to run integration test. The integration service uses the ITS to test application components.
```

**managed workspace** +

> A workspace that mirrors some details of an already existing workspace. Supporting teams can create a managed workspace to grant limited permissions to development teams in the production environment.

Reviewer: ```suggestion
A workspace that mirrors some details of an existing workspace. This allows supporting teams to create a managed workspace to grant limited permissions to development teams in the production environment.
```
